server/base/src/main/java/org/apache/accumulo/server/conf/util/ZooPropEditor.java

```diff
@@ -107,6 +111,10 @@ public void execute(String[] args) throws Exception {
         default:
           throw new IllegalArgumentException("Invalid operation requested");
       }
+    } catch (Exception ex) {
+      LOG.error("{} command failed", keyword(), ex);
+      // hard fail - set exit status
+      System.exit(-1);
```

Comment:
Negative exit codes are not standard. The numbers should be 0 for success, and positive, non-zero for failure. The negation of `0` in binary is `1`, not `-1`. I know we have some existing code that uses negative one, but those should be fixed as well. Eventually all of these should be fixed. There's some discussion about this at https://stackoverflow.com/questions/47265667/return-code-on-failure-positive-or-negative

In particular, if you try to do this in Java, the process exit code is 255, not `-1`. Consider:
```java
public class Test {
  public static void main(String[] args) {
    System.exit(Integer.parseInt(args[0]));
  }
} 
```

Then run it:
```bash
$ for x in {-2..2}; do java Test.java $x; echo $?; done
254
255
0
1
2
```
This means that when you use negative exit codes in Java, you get a positive exit code that is not the same value that you told it to use.

Also, I'm not sure why we're explicitly calling `System.exit` at all. The exception should just fall through the `execute` method into whatever `main` method called it.

Reply:
The point of calling system exit was so that there was an explicit error return status - if some one were to write a script, I wanted to make sure that had an option to test the command return. I was not sure that throwing an exception would do that - and I wanted it to be explicit.